spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationUtils.java

```diff
@@ -59,10 +65,16 @@ static String getDefaultNamespace(String appClassName, int index) {
 	protected static SpringApplicationBuilder embedApp(
 			ConfigurableApplicationContext parentContext, String namespace,
 			Class<?> app) {
-		return new SpringApplicationBuilder(app).web(false).main(app).bannerMode(Mode.OFF)
+		// Child context needs to enable web MVC configuration and web enabled to obtain the MVC request mapping in the child applications
+		return new SpringApplicationBuilder(new Object[]{app, EmbeddedServletContainerAutoConfiguration.class,
+				EndpointWebMvcManagementContextConfiguration.class, WebMvcAutoConfiguration.EnableWebMvcConfiguration.class,
+				ServerPropertiesAutoConfiguration.class})
+				.web(true).main(app)
+				.bannerMode(Mode.OFF)
 				.properties("spring.jmx.default-domain=" + namespace)
 				.properties(
 						InternalPropertyNames.NAMESPACE_PROPERTY_NAME + "=" + namespace)
+				.properties("server.port=-1")
```

Comment:
Still not OK with doing this. If starting one web server per application is the only way we can get this done, then we should drop the idea altogether and not support embedding of web apps.